crm_compassion/model/demand_planning.py

```diff
@@ -0,0 +1,45 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2016 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Maxime Beck
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+
+from openerp import models, fields
+
+
+class WeeklyDemand(models.Model):
+
+    _name = 'demand.weekly.demand'
+    _description = 'Weekly Demand'
+
+    name = fields.Char(required=True)
+    demand_id = fields.Many2one(
+        'demand.planning', 'Demand Planning', readonly=True
+    )
+    week_start_date = fields.Date(required=True)
+    week_end_date = fields.Date(required=True)
+    total_demand = fields.Float(required=True)
+    resupply = fields.Float(required=True)
```

Comment:
Add a field
`period_locked = fields.Boolean()`

that must bet set to True at creation if the week demand is within the next 8 weeks.